dynamic_programming/minimum_edit_distance.cpp

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file
+ * @brief Implementation of [Minimum Edit Distance](https://en.wikipedia.org/wiki/Edit_distance) using Dynamic Programing
+ *
+ * @details
+ *
+ * Given two strings str1 & str2 and we have to calculate the minimum
+ * number of operations (Insert, Remove, Replace) required to convert
+ * str1 to str2.
+ *
+ * ### Algorithm
+ *
+ * We will solve this problem using Naive recursion. But as we are
+ * approaching with a DP solution. So, we will take a DP array to
+ * store the solution of all sub-problems so that we don't have to
+ * perform recursion again and again. Now to solve the problem, We
+ * can traverse all characters from either right side of the strings
+ * or left side. Suppose we will do it from the right side. So, there
+ * are two possibilities for every pair of characters being traversed.
+ * 1. If the last characters of two strings are the same, Ignore
+ * the characters and get the count for the remaining string.
+ * So, we get the solution for lengths m-1 and n-1 in a DP array.
+ *
+ * 2. Else, (If last characters are not the same), we will consider all
+ * three operations (Insert, Remove, Replace) on the last character of
+ * the first string and compute the minimum cost for all three operations
+ * and take the minimum of three values in the DP array.
+ * For Insert: Recur for m and n-1
+ * For Remove: Recur for for m-1 and n
+ * For Replace: Recur for for m-1 and n-1
+ *
+ * @author [Nirjas Jakilim](github.com/nirzak)
+ */
+
+#include <cassert>     /// for assert
+#include <iostream>   /// for IO operations
+#include <vector>    /// for std::vector
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+
+namespace dynamic_programming {
+
+/**
+ * @namespace Minimum Edit Distance
+ * @brief Implementation of [Minimum Edit Distance](https://en.wikipedia.org/wiki/Edit_distance) algorithm
+ */
+
+namespace minimum_edit_distance {
+
+/**
+function to return minimum of three operations
+Insert, Replace or Delete
+*/
+
+int min(int x, int y, int z) {
+  if (x <= y && x <= z) {
+    return x; // return z, if x is the minimum value
+  }
+  if (y <= x && y <= z) {
+    return y; // return y, if y is the minimum value
+  }
+  else {
+    return z; // return z if z is the minimum value
+  }
+}
+
+int editDistDP(std::string str1, std::string str2, int m, int n) {

```

Comment:
This function is missing documentation.

Reply:
I have elaborated the algo on the algorithm section and haven't commented above because this function has plenty of comments inside of it. Just take a look, please. I have given a comment on every section inside of this function so that anyone can properly understand what's happening inside.

Reply:
I know you have put comments inside the function, and that's great: the more documentation we have, the better. But, we also require you to document your function, not only what the function inside does. 🙂